hail/python/hailtop/aiocloud/aiogoogle/client/storage_client.py

```diff
@@ -563,15 +562,20 @@ async def tree_compose(names, dest_name):
                     chunk_tasks = [pool.call(tree_compose, c, n) for c, n in zip(chunks, chunk_names)]
 
                     await pool.wait(chunk_tasks)
+                    await asyncio.gather(*chunk_tasks)  # retrieve exceptions
 
                     await self._compose(chunk_names, dest_name)
 
                     for name in chunk_names:
-                        await pool.call(self._fs._remove_doesnt_exist_ok, f'gs://{self._bucket}/{name}')
+                        await pool.call(
```

Comment:
Why do you need to retrieve exceptions with the gather on line 565, but not after line 572? Is the exception handled somewhere else?